### PR TITLE
Add compass bearing calculations

### PR DESCRIPTION
**Pull request by DarkMatterMatt**

### CODE REVIEW

File: src/context/CompassContext.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+
+/***
+ * Start listening for compass events.
+ *
+ * @param setBearing Accepts a single parameter in the range 0.0-360.0, called when a new bearing is available.
+ * @returns Error message if compass failed to start, or null.
+ */
+async function startCompassListening({ setBearing }) {
```
Comment (HiruNya) [inline src/context/CompassContext.js:9]: I love the documentation!

**Review by f-fromont — APPROVED** [review]
LGTM

File: src/context/CompassContext.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+export const CompassProvider = ({ children }) => {
+  const [bearing, setBearing] = useState(-1);
+  const [errorMessage, setErrorMessage] = useState('Loading...');
+
+  useEffect(() => {
+    startCompassListening({ setBearing }).then((error) => {
+      setErrorMessage(error);
+    });
+  }, []);
```
Comment (HiruNya) [inline src/context/CompassContext.js:90]: Does `setErrorMEssage` have to be in the deps array?

**Review by HiruNya — APPROVED** [review]
LGTM: Have one potential change possible.

**Pull request merged**